java/src/main/bin/build.sh

```diff
@@ -44,18 +44,18 @@ yum install -q -y make epel-release;
 yum install -q -y boost141-devel;
 make clean;
 cat Makefile.permissive | sed 's/BOOST_LIBRARY = -L \/usr\/lib/BOOST_LIBRARY = -L \/usr\/lib64\/boost141/g' | sed 's/BOOST_INCLUDE = -I \/usr\/include/BOOST_INCLUDE = -I \/usr\/include\/boost141/g' > Makefile.permissive.boost141;
-make -f Makefile.permissive.boost141;
+make -f Makefile.permissive.boost141 vw java;
 rm -f Makefile.permissive Makefile.permissive.boost141;
 mv java/target/vw_jni.lib java/target/vw_jni.Red_Hat.5.amd64.lib"
 
 red_hat_6="$early_red_hat
 yum install -q -y boost-devel;
 make clean;
-make -f Makefile.permissive;
+make -f Makefile.permissive vw java;
 rm -f Makefile.permissive;
 mv java/target/vw_jni.lib java/target/vw_jni.Red_Hat.6.amd64.lib"
 
-red_hat_7="yum install -q -y gcc-c++ make boost-devel zlib-devel java-1.7.0-openjdk-devel perl;
+red_hat_7="yum install -q -y gcc-c++ make boost-devel zlib-devel java-1.7.0-openjdk-devel perl clang redhat-lsb-core;
```

Comment:
We need clang because otherwise the redhat 7 build otherwise fails.  Maybe it wasn't caught before because set -e and set -u weren't set in build.sh?  Or am I misunderstanding something?